File: BA StereoSURF/docs/filters-3.3-2009_12/filtersDll-3.3-2009_12/TutorialVisualBasic1/TutorialVisualBasic1/Form1.vb
Imports Filters = FiltersDllDotNet.Filters

Public Class Form1

    Public Sub New()
        InitializeComponent()
        Filters.initialize()
        txtVersion.Text = Filters.getVersion()
        Dim fMax As Int32 = Filters.getFiltersCount()
        For f As Integer = 0 To fMax - 1
            Dim filterName As String = Filters.getFiltersNameAtIndex(f)
            cbFiltersName.Items.Add(filterName)
        Next f

    End Sub

    Protected Overrides Sub Finalize()
        MyBase.Finalize()
        Filters.unInitialize()
    End Sub

    Private Sub cbFiltersName_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cbFiltersName.SelectedIndexChanged
        Dim filterName As String = cbFiltersName.SelectedItem
        ' create a instance of the selected Filter
        Dim filter As Int32 = Filters.createFilter(filterName)
        ' for each parameters of this Filters
        lstParameters.Items.Clear()
        Dim pMax As Int32 = Filters.getParametersCount(filter)
        For p As Integer = 0 To pMax - 1
            ' we get the name and help of parameter
            Dim parameterName As New System.Text.StringBuilder(1024)
            Dim parameterHelp As New System.Text.StringBuilder(1024)
            Filters.getParameterName(filter, p, parameterName)
            Filters.getParameterHelp(filter, p, parameterHelp)
            Dim tmpStr As String = parameterName.ToString + " : " + parameterHelp.ToString
            lstParameters.Items.Add(tmpStr)
        Next p
        ' for each outputs of this Filters
        lstOutputs.Items.Clear()
        Dim oMax As Int32 = Filters.getOutputsCount(filter)
        For o As Integer = 0 To oMax - 1
            ' we get the name of output
            Dim outputName As New System.Text.StringBuilder(1024)
            Filters.getOutputName(filter, o, outputName)
            Dim tmpStr As String = outputName.ToString()
            lstOutputs.Items.Add(tmpStr)
        Next o
        ' we delete this instance of the selected Filter
        Filters.deleteFilter(filter)
    End Sub

End Class

File: BA StereoSURF/docs/filters-3.3-2009_12/filtersDll-3.3-2009_12/TutorialCSharp1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using FiltersDllDotNet;

namespace TutorialCSharp1
{
    public partial class Form1 : Form
    {

        public Form1()
        {
            InitializeComponent();
            Filters.initialize();
            txtVersion.Text = Filters.getVersion();
            Int32 fMax = Filters.getFiltersCount();
            for (int f = 0; f < fMax; f++)
            {
                string filterName = Filters.getFiltersNameAtIndex(f);
                cbFiltersName.Items.Add(filterName);
            }
        }

        ~Form1()
        {
            Filters.unInitialize();
        }

        private void cbFiltersName_SelectedIndexChanged(object sender, EventArgs e)
        {
            string filterName = (string)(cbFiltersName.SelectedItem);
            // create a instance of the selected Filter
            Int32 filter = Filters.createFilter(filterName);
            // for each parameters of this Filters
            lstParameters.Items.Clear();
            Int32 pMax = Filters.getParametersCount(filter);
            for (int p = 0; p < pMax; p++)
            {
                // we get the name and help of parameter
                StringBuilder parameterName = new StringBuilder(1024);
                StringBuilder parameterHelp = new StringBuilder(10240);
                Filters.getParameterName(filter, p, parameterName);
                Filters.getParameterHelp(filter, p, parameterHelp);
                string tmpStr = parameterName + " : " + parameterHelp;
                lstParameters.Items.Add(tmpStr);
            }
            // for each outputs of this Filters
            lstOutputs.Items.Clear();
            Int32 oMax = Filters.getOutputsCount(filter);
            for (int o = 0; o < oMax; o++)
            {
                // we get the name of output
                StringBuilder outputName = new StringBuilder(1024);
                Filters.getOutputName(filter, o, outputName);
                string tmpStr = outputName.ToString();
                lstOutputs.Items.Add(tmpStr);
            }
            // we delete this instance of the selected Filter
            Filters.deleteFilter(filter);
        }

    }
}